manuals: Fix "unusual .Xr" warnings with a script

gperciva opened the pull request.

gperciva commented:
These were reported by `mandoc -T lint ...` as warnings:
- unusual Xr order
- unusual Xr punctuation

The fixes were generated via scripts in the new https://github.com/Tarsnap/freebsd-doc-scripts

1) Generate a list of man pages, excluding contributed code, and runs mandoc on them.
  https://github.com/Tarsnap/freebsd-doc-scripts/blob/main/generate-man-problem-lists.sh#L16

2) Run a python script on those files to sort & punctuate the SEE ALSO sections.
  https://github.com/Tarsnap/freebsd-doc-scripts/blob/main/fixes.py#L4

  Ignoring a few corner cases, the script is essentially just:
  ```
  xrs = man.get_section("SEE ALSO")

  # strip commas
  xrs = [xr[:-2] if xr.endswith(" ,") else xr for xr in xrs]

  # sort the way mandoc wants it
  xrs.sort(key=lambda x: (x.split()[2], x.split()[1].lower()))

  # add back commas as appropriate
  xrs[:-1] = ["%s ," % x for x in xrs[:-1]]
  man.replace_section("SEE ALSO", xrs)
  ```

I fully expect more problems to appear in the SEE ALSO section over time, so I'm planning on running the script every month or so.  Also, I'll be expanding the script to include other "low-hanging fruit" it can fix.  Licensed as BSD 2-clause, so if there's ever a desire to move it to the FreeBSD repo itself, that's easily done.

@mhorne, @concussious

concussious reviewed (commented):
Reading over the changes, the reordering looks great except for the mentioned memccpy twice.

This does not constitute checking if those pages still exist, or reading the manual to identify if these are the right pages to be linking to, or if the linked name is the most appropriate name to link to, or if other pages are appropriate to link to. I like to actually skim every xref, which seems impossible to do this way.

So, the reordering is good, but this doesn't constitute a review of the actual information in the manual, so the dates shouldn't be bumped (which, you didn't).

I'd like if your commit message mentioned that? Something like:
```
manuals: automated reorder of see also
```

gperciva commented on lib/libc/string/bstring.3:
```diff
@@ -81,9 +81,9 @@ See the specific manual pages for more information.
 .Xr bcopy 3 ,
 .Xr bzero 3 ,
 .Xr memccpy 3 ,
+.Xr memccpy 3 ,
```
Good catch!  ... huh, `main` currently has:
```
.Xr memccpy 3 ,
.Xr memchr 3 ,
.Xr memcmp 3 ,
.Xr memccpy 3 ,
.Xr memcpy 3 ,
```

(with two `memccpy`)

I'll switch the script to use a `set()` instead of a list, so that it eliminates any duplicate lines.  Give me a few minutes.

gperciva commented:
Duplicate `memccpy` fixed.  No other duplicates found by the script.

I'm happy to adjust to the subject line if @mhorne thinks it should be.  My initial thought was that it wasn't worth mentioning a script until/unless that script was in `-CURRENT`... and it'll probably be months until the script is sufficiently well-tested & trusted for that to make sense.

> This does not constitute checking if those pages still exist

`mandoc -T lint` takes care of that... although it's a STYLE warning, rather than WARNING.

Out of the 79 files changed by this commit, there's 34 `.Xr` lines which point to missing manuals.  Not great, but not terrible... and I'd say that fixing those is outside the scope of this PR.

A quick glance at the first 6 such errors (and testing with `apropos`) shows that 3 of them simply have an incorrect section number, and 1 more is missing a prefix.  For example, `.Xr cd9660 4` is actually in section `5`.  That's something which could be fixed with a script, but if these man pages are representative, there's going to be around thousands of fixes required.  Reviewing those would be a massive pain.

> I like to actually skim every xref, which seems impossible to do this way.

I'd say that 4000+ man pages, each with multiple xrefs, are impossible to skim, no matter what way.  :)  That said, it's certainly worth manually checking the most important man pages (and you can define "important" in this context however you want!).

My goal is to work in conjunction with the already-trusted tool -- `mdoc`, mentioned in the contributor docs -- to clear up easy stuff.  If a tool gives you 2400 warnings & errors (of which 90% are trivial), it's easy to miss a few important problems.  Whereas if that tool only shows 24 warnings, people are much more likely to take them seriously.

concussious reviewed (approved):
This is very cool! Thanks for tagging me!

mhorne commented:
> I'm happy to adjust to the subject line if @mhorne thinks it should be. My initial thought was that it wasn't worth mentioning a script until/unless that script was in `-CURRENT`... and it'll probably be months until the script is sufficiently well-tested & trusted for that to make sense.
> 

It is your choice if you wish to mention it or not. It is somewhat useful to note that the changes were generated programmatically, but communicating this fact alone does not oblige you to ever finish or release the scripts.

emaste commented:
I agree that it's worth mentioning the script, even if it's not yet committed (or even won't ever be).

gperciva commented:
Ok, added "with a script" to the subject, and dropped a link to the script's github repo in the comment body.

The repository is subject to change as the scripts get more complicated (e.g., putting things in directories, adding a `pyproject.toml`), but I'd hope that anybody interested in digging into the software archaeology would know to checkout that repository at the appropriate date.
